rotkehlchen/chain/evm/decoding/pendle/decoder.py

```diff
@@ -280,6 +294,149 @@ def _decode_pendle_swap(self, context: DecoderContext) -> DecodingOutput:
+            return DEFAULT_DECODING_OUTPUT
+
+        maybe_reshuffle_events(
+            ordered_events=[out_event] + in_events,
+            events_list=context.decoded_events,
+        )
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_redeem_sy_pt_yt_to_token(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes Pendle redemptions where SY, PT, or YT are converted back to the underlying token."""  # noqa: E501
+        amount_in = asset_normalized_value(
+            amount=int.from_bytes(context.tx_log.data[64:96]),
+            asset=(token_in := self.base.get_token_or_native(bytes_to_address(context.tx_log.topics[2]))),  # noqa: E501
+        )
+        amount_out = asset_normalized_value(
+            amount=int.from_bytes(context.tx_log.data[32:64]),
+            asset=self.base.get_token_or_native(yt_address := bytes_to_address(context.tx_log.topics[3])),  # noqa: E501
+        )
+        out_events, in_event = [], None
+        for event in context.decoded_events:
+            if (
+                    # Pendle emits one SPEND per output token (SY, PT, YT),
+                    # each with the same amount.
+                    event.amount == amount_out and
+                    event.address == yt_address and
+                    event.event_type in (HistoryEventType.SPEND, HistoryEventType.TRADE) and
+                    event.event_subtype in (HistoryEventSubType.NONE, HistoryEventSubType.SPEND)
+            ):
+                out_events.append(event)
+                event.counterparty = CPT_PENDLE
+                event.event_type = HistoryEventType.SPEND
+                event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                event.notes = f'Return {event.amount} {event.asset.resolve_to_asset_with_symbol().symbol} to Pendle'  # noqa: E501
+
+            elif (
+                    event.amount == amount_in and
+                    event.asset == token_in and
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE
+            ):
+                in_event = event
+                event.counterparty = CPT_PENDLE
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REDEEM_WRAPPED
+                event.notes = f'Withdraw {event.amount} {token_in.symbol} from Pendle'
+
+        if in_event is None or len(out_events) == 0:
+            log.error(f'Could not retrieve either out events or in event in pendle redeem transaction {context.transaction}')  # noqa: E501
+            return DEFAULT_DECODING_OUTPUT
+
+        maybe_reshuffle_events(
+            ordered_events=out_events + [in_event],
+            events_list=context.decoded_events,
+        )
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_exit_post_exp_to_token(self, context: DecoderContext) -> DecodingOutput:
```

Comment:
same for this function. It's literally copy pasting stuff with a few different changes in some variables